test.rs

```diff
@@ -0,0 +1,41 @@
+mod common;
+
+// Include tests directly in this file if desired
+#[cfg(test)]
+mod tests {
+    // use rust_vmmap::*;
+
+    use crate::common;
+
+    #[test]
+    fn tests() {
+        common::setup();
+        //TODO: implement integration test
+
+        let mut vmmap = Vmmap::new();
+        
+        // Test adding a valid entry (should pass)
+        let entry = VmmapEntry::new(0, 100, PROT_READ | PROT_WRITE, 0);
+        assert!(vmmap.add_entry(entry).is_ok());
```

Comment:
So the current add_entry needs a bit of work because we expect a vmmap entry directly. 

However normally we would use add_entry_with_override, which accepts all the parameters and constructs it's own vmmap. But this is kind of a pain, so i had left in the "stub" for add_entry. I think for now switch these tests to use add_entry_with_override instead of this